app/javascript/components/select/Selector.vue

```diff
@@ -0,0 +1,200 @@
+<template>
+  <div class="selector">
+    <div
+      class="selector__selected"
+      ref="selected"
+      tabindex="0"
+      @focus="showDropdown(true)"
+      @click="showDropdown(true)"
+      @keyup="onKeyup"
+      @keyup.enter="showDropdown(true)"
+    >
+      <div class="selector__label">{{ state.label }}</div>
+      <div
+        class="selector__caret"
+        :class="{
+            'selector__caret--active': dropdownEnabled
+        }"
+      />
+      <div v-if="dropdownEnabled" class="selector__options">
+        <div
+          class="selector__option"
+          ref="options"
+          tabindex="0"
+          v-for="opt in computedOptions.slice(1)"
+          :class="{ 'selector__option--active': isSelected(opt.value) }"
+          :key="opt.value"
+          @click="select(opt.value)"
+          @keyup="onKeyup"
+          @keyup.enter.stop="select(opt.value)"
+          @keyup.esc.stop="showDropdown(false)"
+          @keyup.up.prevent.stop="onKeyArrowUp(opt)"
+          @keyup.down.prevent.stop="onKeyArrowDown(opt)"
+        >{{ opt.label }}</div>
+      </div>
+    </div>
+  </div>
+</template>
+<script>
+import { delay, debounce, intersectionBy } from "lodash";
+
+export default {
+  model: {
+    event: "change",
+    prop: "state"
+  },
+  props: {
+    default: {
+      required: false
+    },
+    state: {
```

Comment:
state could could perhaps be renamed slightly to make it clearer that it's the selected value (if it that is what it is)